client/js/render.js

```diff
@@ -15,57 +15,85 @@ const sidebar = $("#sidebar");
 module.exports = {
 	appendMessage,
 	buildChannelMessages,
-	buildChatMessage,
 	renderChannel,
-	renderChannelMessages,
 	renderChannelUsers,
 	renderNetworks,
 };
 
-function buildChannelMessages(data) {
-	return data.messages.reduce(function(docFragment, message) {
-		appendMessage(docFragment, data.id, data.type, message.type, buildChatMessage({
-			chan: data.id,
-			msg: message
-		}));
+function buildChannelMessages(chanId, chanType, messages) {
+	return messages.reduce(function(docFragment, message) {
+		appendMessage(docFragment, chanId, chanType, message);
 		return docFragment;
 	}, $(document.createDocumentFragment()));
 }
 
-function appendMessage(container, chan, chanType, messageType, msg) {
+function appendMessage(container, chanId, chanType, msg) {
+	const renderedMessage = buildChatMessage(chanId, msg);
+
+	// Check if date changed
+	let lastChild = container.find(".msg").last();
+	const msgTime = new Date(msg.time);
+
+	// It's the first message in a window,
+	// then just append the message and do nothing else
```

Comment:
An idea for a future PR would be to display the date marker if it's the first message in a window **and** there is no "more" button available. Reason for this would be to remove that odd situation that #1318 shows (I know it's not creating it, just making it more obvious).